### PR TITLE
aws - iam certificate - add delete action

### PR DESCRIPTION
**Pull request by dnouri**
Consider the `action` in this example:

```yaml
        - name: aws-iam-certificate-delete-expired
          resource: iam-certificate
          filters:
            - type: value
              key: Expiration
              value_type: expiration
              op: greater-than
              value: 0
          actions:
            - type: delete
```

### CODE REVIEW

File: c7n/resources/iam.py [hunk 1]
```diff
@@ -440,6 +440,88 @@ class resource_type(TypeInfo):
...
+    def process(self, resources):
+
+        # get reference to AWS client
+        client = local_session(self.manager.session_factory).client('iam')
+
+        # for each resource call delete
+        for cert in resources:
+            self.manager.retry(
+                client.delete_server_certificate,
+                ServerCertificateName=cert['ServerCertificateName'],
```
Comment (kapilt) [inline c7n/resources/iam.py:481]: pass ignore_err_code to not raise for certificates already deleted/in progress on deletion.

File: c7n/resources/iam.py [hunk 2]
```diff
@@ -440,6 +440,88 @@ class resource_type(TypeInfo):
...
+            )
+
+# {
+#     "Path": "/",
```
Comment (kapilt) [inline c7n/resources/iam.py:485]: this looks like noise, please remove.

File: tests/test_iam.py [hunk 3]
```diff
@@ -1055,6 +1055,68 @@ def test_iam_group_delete(test, iam_user_group):
...
+@terraform('iam_delete_certificate', teardown=terraform.TEARDOWN_IGNORE)
+def test_iam_delete_certificate_action(test, iam_delete_certificate):
+    iam_name = iam_delete_certificate[
+        'aws_iam_server_certificate.test_cert_alt.name']
+
+    session_factory = test.replay_flight_data('iam_delete_certificate')
+    client = session_factory().client('iam')
+
+    try:
+        client.get_server_certificate(ServerCertificateName=iam_name)
```
Comment (kapilt) [inline tests/test_iam.py:1067]: remove this api call, and add the assertion on server cert name to the policy as an attribute filter

**Review by kapilt — CHANGES_REQUESTED** [review]
comments inline

**Review by kapilt — APPROVED** [review]
lgtm

**Review by marcoceppi — APPROVED** [review]
lgtm

**Pull request merged**